realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsRemoteMongoCollection.cpp

```diff
@@ -33,13 +33,36 @@
 
 using namespace realm;
 using namespace realm::app;
+using namespace realm::bson;
 using namespace realm::jni_util;
 using namespace realm::_impl;
 
-static std::function<jobject(JNIEnv*, uint64_t)> collection_mapper = [](JNIEnv* env, uint64_t result) {
+// This mapper works for both count and delete operations
+static std::function<jobject(JNIEnv*, uint64_t)> collection_mapper_count = [](JNIEnv* env, uint64_t result) {
     return JavaClassGlobalDef::new_long(env, result);
 };
 
+static std::function<jobject(JNIEnv*, util::Optional<bson::BsonDocument>)> collection_mapper_find_one = [](JNIEnv* env, util::Optional<bson::BsonDocument> document) {
+    return document ? JniBsonProtocol::bson_to_jstring(env, *document) : NULL;
+};
+
+static std::function<jobject(JNIEnv*, util::Optional<ObjectId>)> collection_mapper_insert_one = [](JNIEnv* env, util::Optional<ObjectId> object_id) {
+    return JavaClassGlobalDef::new_object_id(env, object_id.value());
+};
+
+static std::function<jobject(JNIEnv*, std::vector<ObjectId>)> collection_mapper_insert_many = [](JNIEnv* env, std::vector<ObjectId> object_ids) {
+    jobjectArray arr = (jobjectArray)env->NewObjectArray(static_cast<jsize>(object_ids.size()), JavaClassGlobalDef::java_lang_object(), NULL);
```

Comment:
This isn't an `Optional`. What I could do instead is to check whether the array is empty or not. In case it's empty then we can throw an exception, although it should never reach that point in here, as the only scenario I can think of would generate an empty array is a failure in inserting the documents, which would trigger an error callback.